### PR TITLE
[DEV-1062] Add confirmation page

### PR DESCRIPTION
**Pull request by MarcoPonchia**
#### List of Changes

Add confirmation page that evaluate username and code through congnito API after sign-up and confirm user if valid

#### Motivation and Context

Allow user confirmation through link in email 

#### How Has This Been Tested?

Creating a new user and navigate on http://localhost:3000/auth/confirmation?username=XXX&code=YYY using instead of XXX and YYY username and code received in email if valid you should redirected to `/auth/account-activated` if invalid you should stay on confirmation route and see 404 error

#### Screenshots
[Screencast from 31-10-2023 18:26:52.webm](https://github.com/pagopa/developer-portal/assets/24475319/55f27233-ef3b-4688-9716-552d08cc0962)

#### Types of changes

- [x] Chore (nothing changes by a user perspective)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: apps/nextjs-website/src/app/auth/confirmation/page.tsx [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+        .catch(() => {
+          setRenderNotFound(true);
+        });
+    } else {
+      setRenderNotFound(true);
+    }
+  }, [username, code]);
+
+  return renderNotFound ? (
+    <PageNotFound />
```
Comment (marcobottaro) [inline apps/nextjs-website/src/app/auth/confirmation/page.tsx:34]: This should be replaced with a more specific error page. I think we should add a TODO

File: apps/nextjs-website/src/app/auth/confirmation/page.tsx [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+import { useSearchParams, useRouter } from 'next/navigation';
+import { useEffect, useState } from 'react';
+
+const Confirmation = () => {
+  const searchParams = useSearchParams();
+  const router = useRouter();
+  const username = searchParams.get('username');
+  const code = searchParams.get('code');
+
+  const [renderNotFound, setRenderNotFound] = useState(environment === 'prod');
```
Comment (kin0992) [inline apps/nextjs-website/src/app/auth/confirmation/page.tsx:15]:
```suggestion
  const [renderNotFound, setRenderNotFound] = useState(isProduction);
```
I think here you could use the `isProduction` comring from the `@/config`

File: apps/nextjs-website/src/app/auth/confirmation/page.tsx [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+    if (username && code) {
+      Auth.confirmSignUp(username, code)
+        .then(() => {
+          setRenderNotFound(false);
+          // eslint-disable-next-line functional/immutable-data
+          router.push('/auth/account-activated');
+        })
+        .catch(() => {
+          setRenderNotFound(true);
+        });
```
Comment (datalek) [inline apps/nextjs-website/src/app/auth/confirmation/page.tsx:27]: What about to handle errors in a more specific way? If any error occurs the `configSignUp` endpoint returns a response like the following one:

```
{"name":"ExpiredCodeException","code":"ExpiredCodeException"}
```

You can find more details reading the [cognito docs](https://docs.aws.amazon.com/cognito-user-identity-pools/latest/APIReference/API_ConfirmSignUp.html#API_ConfirmSignUp_Errors).

Choose yourself between create a ticket about a better error handling or do the improve now

Comment (MarcoPonchia) [reply apps/nextjs-website/src/app/auth/confirmation/page.tsx:27]: We add this [ticket](https://pagopa.atlassian.net/browse/DEV-1047) to handle mange login and signup errors

**Pull request merged**